### PR TITLE
ci: trim permissions for codeql-analysis action

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Trim permissions for `codeql-analysis.yml`

## Context:

I think we can tighten down our permissions for the CodeQL action as well.
There's no official documentation in their README yet, but I did find this issue: https://github.com/github/codeql-action/issues/464

Please read the linked issue before reviewing my PR, maybe I'm missing something important from that discussion!

In that issue a contributor to the project says:

> As far as I can tell, if you are using the default workflow, you _should_ only need the following permissions:
> 
> ```
>     permissions:
>       contents: read
>       security-events: write
>       pull-requests: read
> ```
> 
> Is your workflow doing anything special?

I figured, we can try these permissions.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -9,6 +9,11 @@ on:
   schedule:
     - cron: '0 13 * * 1'
 
+permissions:
+  contents: read
+  security-events: write
+  pull-requests: read
```
Comment (HonkingGoose) [inline .github/workflows/codeql-analysis.yml:15]: Hmm, maybe we should specify the permissions within each `job` entry instead of globally...

**Comment by viceice** [comment]
I'm not sure if we really need this, as this are all primary actions 🤔

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (viceice) [reply .github/workflows/codeql-analysis.yml:15]: it doesn't matter, as we have only one job

**Review by rarkins — DISMISSED** [review]

**Comment by HonkingGoose** [comment]
> I'm not sure if we really need this, as this are all primary actions 🤔

What do you think @rarkins?

**Review by viceice — DISMISSED** [review]

**Comment by rarkins** [comment]
It's still best practice, but hopefully never needed

**Comment by HonkingGoose** [comment]
I'd say we're ready to merge and see what happens, I think this is OK, but you can never be sure until you actually run it...

---

**If** the action breaks somehow after the merge, we can try out this suggestion from a collaborator:

> It's possible that you will also need the: `actions: read` permission. Some code flows will make requests to introspect the current workflow and this permission is needed. So, if you get any more failures, try adding that permission as well.

https://github.com/github/codeql-action/issues/464#issuecomment-828831286

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (viceice) [inline .github/workflows/codeql-analysis.yml:15]:
```suggestion
  security-events: write
```

It seems to that this is enough

**Review by viceice — APPROVED** [review]

**Pull request merged**